runhouse/servers/env_servlet.py

```diff
@@ -227,61 +254,100 @@ def _get_env_cpu_usage(self, cluster_config: dict = None):
         except psutil.NoSuchProcess:
             env_memory_usage = {}
 
-        return (env_memory_usage, node_name, total_memory, env_servlet_pid, node_ip)
+        return (env_memory_usage, node_name, total_memory, self.pid, node_ip)
 
-    def _get_env_gpu_usage(self, env_servlet_pid: int):
-        import subprocess
+    def _get_env_gpu_usage(self):
+        # currently works correctly for a single node GPU. Multinode-clusters will be supported shortly.
 
-        try:
+        collected_gpus_info = copy.deepcopy(self.gpu_info)
 
-            gpu_general_info = (
-                subprocess.run(
-                    [
-                        "nvidia-smi",
-                        "--query-gpu=memory.total",
-                        "--format=csv,noheader,nounits",
-                    ],
-                    stdout=subprocess.PIPE,
-                )
-                .stdout.decode("utf-8")
-                .strip()
-                .split(", ")
-            )
-            total_gpu_memory = int(gpu_general_info[0]) * (1024**2)  # in bytes
-
-            env_used_memory = 0  # in bytes
-
-            env_gpu_usage = (
-                subprocess.run(
-                    [
-                        "nvidia-smi",
-                        "--query-compute-apps=pid,gpu_uuid,used_memory",
-                        "--format=csv,nounits",
-                    ],
-                    stdout=subprocess.PIPE,
-                )
-                .stdout.decode("utf-8")
-                .strip()
-                .split("\n")
+        if collected_gpus_info is None or not collected_gpus_info[0]:
+            return None
+
+        gpus = list(collected_gpus_info.keys())
```

Comment:
Looks like the code below here is duplicated in the _aperiodic... method. Maybe it could be abstracted into a method to use in both places?